pkg/controllers/allocation/scheduling/suite_test.go

```diff
@@ -368,6 +368,91 @@ var _ = Describe("NodeAffinity", func() {
 	})
 })
 
+var _ = Describe("Preferential Fallback", func() {
+	Context("Required", func() {
+		It("should not relax requirements", func() {
+			provisioner.Spec.Zones = []string{"test-zone-1"}
+			provisioner.Spec.InstanceTypes = []string{"default-instance-type"}
+			pod := test.UnschedulablePod()
+			pod.Spec.Affinity = &v1.Affinity{NodeAffinity: &v1.NodeAffinity{RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{NodeSelectorTerms: []v1.NodeSelectorTerm{
+				{MatchExpressions: []v1.NodeSelectorRequirement{
+					{Key: v1.LabelTopologyZone, Operator: v1.NodeSelectorOpIn, Values: []string{"invalid"}}, // Should not be relaxed
```

Comment:
Just checking - it should not be relaxed because it's the last (only) one?

Reply:
Correct